src/controller.js

```diff
@@ -0,0 +1,126 @@
+(function exportController() { 
+
+  class Controller {
+    constructor(ship) {
+      this.ship = ship;
+      this.initialiseSea();
+      document.querySelector("#addbutton").addEventListener("click", () => {
+        this.addPort();
+        this.renderPorts(ship.itinerary.ports);
+      });
+      document.querySelector("#sailbutton").addEventListener("click", () => {
+        this.setSail();
+        document.querySelector("#sailbutton").disabled = true;
+        document.querySelector("#addbutton").disabled = true;
```

Comment:
I would also move these into `setSail`, as that is where you're re-enabling the `sailButton`, so makes it easier to read